src/kudu/consensus/consensus_queue.cc

```diff
@@ -999,8 +999,62 @@ void PeerMessageQueue::AdvanceMajorityReplicatedWatermarkFlexiRaft(
 }
 
 std::map<std::string, int> PeerMessageQueue::GetDataCommitQuorum() {
+  // Map to store the number of voters in each region from the active config.
+  std::map<std::string, int> voter_distribution;
```

Comment:
this function does not seem to be thread safe? it doesn't take the queue lock.

Reply:
Line 1313 seems to take the queue lock before calling this. Also, this function is read-only. I have marked this as constant.